frontend/src/components/charts/DurationSummaryChart.tsx

```diff
@@ -0,0 +1,128 @@
+import { useTheme } from "@mui/material/styles/";
+import ReactEcharts from "echarts-for-react";
+
+import { Difficulty } from "~shared/types/base/index";
+
+type DurationSummaryProps = {
+  durationSummary:
+    | {
+        difficulty: Difficulty;
+        timetaken: number;
+        date: Date;
+      }[]
+    | undefined;
+};
+
+export const DurationSummaryChart = ({
+  durationSummary,
+}: DurationSummaryProps) => {
+  const theme = useTheme();
+
+  const processDates = (
+    durationSummary:
+      | {
+          difficulty: Difficulty;
+          timetaken: number;
+          date: Date;
+        }[]
+      | undefined,
+  ) => {
+    const xAxis = new Set<string>();
+    if (!durationSummary) {
+      return;
+    } else {
+      // Sort the durationSummary first
+      durationSummary.sort((a, b) => a.date.getTime() - b.date.getTime());
```

Comment:
Dates aren't deserialised automatically when being received by the frontend.

Reply:
To use Date | string in dto